lib/IRGen/IRGenModule.cpp

```diff
@@ -621,6 +621,7 @@ IRGenModule::IRGenModule(IRGenerator &irgen,
   AsyncFunctionPointerPtrTy = AsyncFunctionPointerTy->getPointerTo(DefaultAS);
   SwiftTaskPtrTy = SwiftTaskTy->getPointerTo(DefaultAS);
   SwiftAsyncLetPtrTy = Int8PtrTy; // we pass it opaquely (AsyncLet*)
+  SwiftTaskOptionRecordPtrTy = Int64Ty; // Builtin.RawPointer? that we get as (TaskOptionRecord*)
```

Comment:
Why not `Int8PtrTy` here, like with `SwiftAsyncLetPtrTy`?

Reply:
Hah yeah this is what took me a week to figure out weirdly enough... With the kind of some folks we figured out that the lowering of the `Optional<Builtin.RawPointer>` ends up as int64 and we must have those two align. I indeed wanted `i8*` here but there didn't seem to be a way to solve it AFAICS -- I'm sure these is some way tho